packages/protocol/src/did-resolver.ts

```
@@ -47,7 +48,7 @@ export type DidResource = DidDocument | VerificationMethod | DidService
  * @throws if DID cannot be resolved
  * @beta
  */
-export async function deferenceDidUrl(didUrl: string): Promise<DidResource> {
+export async function deferenceDidUrl(didUrl: string): Promise<DidResource | undefined> {
```

How should `deferenceDidUrl()` behave if verification methods or services are missing? Should we throw or is returning undefined sufficient?

---

@andresuribe87 Any thoughts here?

---

Given that DID URL dereferencing is defined in https://www.w3.org/TR/did-core/#did-url-dereferencing, I would suggest that this function should return something similar to what's defined in the spec. 

If that's the case, then I can see the case for returning an object similar to the example in https://www.w3.org/TR/did-core/#did-url-dereferencing. In particular, for the example mentioned, the property named `didUrlDereferencingMetadata`, would have an object which contains an `error` property. The value of the property would be `notFound`.

---

Ah, looks like web5/dids has an implementation of DID URL dereferencing but it's not in the latest version. I'll just mark a TODO #147